src/FsAutoComplete.Core/UntypedAstUtils.fs

```diff
@@ -1,668 +1,663 @@
+    | SynArgPats.Pats ps -> ps
+    | SynArgPats.NamePatPairs(pats = xs) -> xs |> List.map (fun (_, _, pat) -> pat)
+
+  /// A pattern that collects all patterns from a `SynSimplePats` into a single flat list
+  let (|AllSimplePats|) (pats: SynSimplePats) =
+    let rec loop acc pat =
+      match pat with
+      | SynSimplePats.SimplePats(pats = pats) -> acc @ pats
+
+    loop [] pats
+
+  type SyntaxCollectorBase() =
+    abstract WalkSynModuleOrNamespace: SynModuleOrNamespace -> unit
+    default _.WalkSynModuleOrNamespace m = ()
+    abstract WalkAttribute: SynAttribute -> unit
+    default _.WalkAttribute a = ()
+    abstract WalkSynModuleDecl: SynModuleDecl -> unit
+    default _.WalkSynModuleDecl m = ()
+    abstract WalkExpr: SynExpr -> unit
+    default _.WalkExpr s = ()
+    abstract WalkTypar: SynTypar -> unit
+    default _.WalkTypar s = ()
+    abstract WalkTyparDecl: SynTyparDecl -> unit
+    default _.WalkTyparDecl s = ()
+    abstract WalkTypeConstraint: SynTypeConstraint -> unit
+    default _.WalkTypeConstraint s = ()
+    abstract WalkType: SynType -> unit
+    default _.WalkType s = ()
+    abstract WalkMemberSig: SynMemberSig -> unit
+    default _.WalkMemberSig s = ()
+    abstract WalkPat: SynPat -> unit
+    default _.WalkPat s = ()
+    abstract WalkValTyparDecls: SynValTyparDecls -> unit
+    default _.WalkValTyparDecls s = ()
+    abstract WalkBinding: SynBinding -> unit
+    default _.WalkBinding s = ()
+    abstract WalkSimplePat: SynSimplePat -> unit
+    default _.WalkSimplePat s = ()
+    abstract WalkInterfaceImpl: SynInterfaceImpl -> unit
+    default _.WalkInterfaceImpl s = ()
+    abstract WalkClause: SynMatchClause -> unit
+    default _.WalkClause s = ()
+    abstract WalkInterpolatedStringPart: SynInterpolatedStringPart -> unit
+    default _.WalkInterpolatedStringPart s = ()
+    abstract WalkMeasure: SynMeasure -> unit
+    default _.WalkMeasure s = ()
+    abstract WalkComponentInfo: SynComponentInfo -> unit
+    default _.WalkComponentInfo s = ()
+    abstract WalkTypeDefnSigRepr: SynTypeDefnSigRepr -> unit
+    default _.WalkTypeDefnSigRepr s = ()
+    abstract WalkUnionCaseType: SynUnionCaseKind -> unit
+    default _.WalkUnionCaseType s = ()
+    abstract WalkEnumCase: SynEnumCase -> unit
+    default _.WalkEnumCase s = ()
+    abstract WalkField: SynField -> unit
+    default _.WalkField s = ()
+    abstract WalkTypeDefnSimple: SynTypeDefnSimpleRepr -> unit
+    default _.WalkTypeDefnSimple s = ()
+    abstract WalkValSig: SynValSig -> unit
+    default _.WalkValSig s = ()
+    abstract WalkMember: SynMemberDefn -> unit
+    default _.WalkMember s = ()
+    abstract WalkUnionCase: SynUnionCase -> unit
+    default _.WalkUnionCase s = ()
+    abstract WalkTypeDefnRepr: SynTypeDefnRepr -> unit
+    default _.WalkTypeDefnRepr s = ()
+    abstract WalkTypeDefn: SynTypeDefn -> unit
+    default _.WalkTypeDefn s = ()
+
+  let walkAst (walker: SyntaxCollectorBase) (input: ParsedInput) : unit =
```

Comment:
The implementation here is the same as before, but with all of the `addIfInside` calls replace with calls to `walker.WalkX` as appropriate.